geo: rebrand geospatial to spatial with minor refactors

otan opened the pull request:
* Rename geospatial to spatial (for outwards facing docs).
* Replace all (args[d].(*tree.DType)) to tree.MustBeDType(args[d])
* Add tests to ensure all geo builtin info has a capital letter and
  full stop.

Release justification: low risk update to new functionality

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53721)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 11 of 11 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53721#-:-MG9IZPJ9vuzwy62g0Nr:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/sql/sem/builtins/geo_builtins.go, line 51 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/53721#-MG9HReO2UIhwvKWkd0F:-MG9HReO2UIhwvKWkd0G:b-2c2v49) ([raw file](https://github.com/cockroachdb/cockroach/blob/325ad66b52cc1751f8d185809f4221d6eba9db8e/pkg/sql/sem/builtins/geo_builtins.go#L51)):*
> ```Go
> const usesSpheroidMessage = " Uses a spheroid to perform the operation."
> const spheroidDistanceMessage = `"\n\nWhen operating on a spheroid, this function will use the sphere to calculate ` +
> 	`the closest two points using a sphere. The spheroid distance between these two points is calculated using GeographicLib. ` +
> ```

I think we can end the sentence at "... closest two points." since it already says "will use the sphere" earlier in the sentence.


<!-- Sent from Reviewable.io -->

otan committed:
geo: rebrand geospatial to spatial with minor refactors

* Rename geospatial to spatial (for outwards facing docs).
* Replace all (args[d].(*tree.DType)) to tree.MustBeDType(args[d])
* Add tests to ensure all geo builtin info has a capital letter and
  full stop.

Release justification: low risk update to new functionality

Release note (cli change): The `--geo-libs` flag introduced in
v20.2-alphas is renamed to `--spatial-libs`.

otan commented:
bors r=sumeerbhola
